e2e/test/framework/pod.go

```diff
@@ -42,6 +38,26 @@ func GetPodObject(name, namespace, pvc string) *core.Pod {
 			},
 		},
 	}
+
+	if volumeType == core.PersistentVolumeFilesystem {
```

Comment:
Exhaustive switch instead of two `if`s?

Reply:
the thing I know i wanted is passing in a required non-pointer volume type to pod creation and this accomplished that so we can write a `Context()` to set volumeType like [this](https://github.com/linode/linode-blockstorage-csi-driver/pull/140/files#diff-7bc5b2d13fd2086f3a76c2fc154cfdefc57ed7f8e3964031c5c8936fa76c2876R273). can we leave this for the moment? i believe a follow up task to add to the block tests to have some setup that makes them behave like the filesystem is coming and this will be refactored in the process. if i refactored this today it'll just be a switch with two cases and if you want that let me know.

Reply:
I just want this to err (?) if it's not one of these two types as that seems unsupported (?)

Reply:
that's a totally fair ask. i rewrote as a switch and added error returns for both the pod and pvc funcs and made them look/behave similarly so they feel more cohesive. let me know if you approve these changes: https://github.com/linode/linode-blockstorage-csi-driver/pull/140/commits/c63960a2a0a1b04a6bcc18c8b54e1580f0928a8e